### DISCUSSION TITLE
overridable classes

### DISCUSSION THREAD

**Discussion by daalfox** [tailwindlabs/tailwindcss#10882] [Ideas]
Is it possible to achieve something like `weak:bg-neutral-400` that will be overridden by any kind of overriding? This can be very useful in library development where you want to add a class, but you want the users to be able to override this easily without using `!important`

**Comment by wongjn** [comment]
The solutions that come to mind are:

## [`:where()`](https://developer.mozilla.org/en-US/docs/Web/CSS/:where)

The `where()` pseudo-class can be used to select an element with `(0 0 0)` specificity, thus making it easily overridable:

```js
addVariant('weak', ':where(&)');
```

## [`@layer`](https://developer.mozilla.org/en-US/docs/Web/CSS/@layer)

The `@layer` at-rule can be used to group rules, and order these groups according to the desired precedence. Rules in a `@layer` will always be overridden by rules not in a `@layer`. Not to be confused with Tailwind's `@layer base/components/utilities` directives.

```js
addVariant('weak', '@layer weak');
```

## Example

[Here's a Tailwind Play example showcasing both techniques](https://play.tailwindcss.com/dsztV9cDtD).
